### ISSUE TITLE
Using isOver when child has returned canDrop = false

### ISSUE THREAD

**Issue by brtdv** [react-dnd/react-dnd#1484]
Hi guys,

Please consider the following example:
https://codesandbox.io/s/flamboyant-pascal-yru3y?fontsize=14, or see also this video with a summary of the problem: https://imgur.com/a/GrHK8q8

I have multiple `DragItem` components, on the page the nested drop targets `Box` with some `InnerBox` components.

Every DragItem component has a `[type]` property, (that I have limited for this example into only 2 types). I have DragItem components props `[type=Type1]` and `[type=Type2]`. These DragItem components can only be dropped into the `InnerBox` components with the same `type` property. The outer, `Box` dropzone, accepts all `DragItem` components, no matter the type.

For the 'event handling' side, when the DragItem was dragged on an `InnerBox`, only the `InnerBox` compoment should trigger some function. In my above example, this works perfectly, by using the `drop()` function. In the `drop()` function of `Box`, I can use `monitor.didDrop()` as described in the documentation, to check if the item was already dropped on the child.

However, I seem to be stuck with the use of the monitor.isOver(), monitor.isOverCurrent().

When you drag a `DragItem[type=Type1]` over the `InnerBox[type=Type1]`, the `InnerBox[type=Type1]` is highlighted. Same when I drag over the `DragItem` over the `Box` component. However: how would could I make sure the the `Box` component gets a hover state by using isOver / isOverCurrent when the `DragItem[type=Type1]` is hovered over `InnerBox[type=Type2]`? 

I seem to be stuck, I have no idea how I could make sure that the isOver property of Box is `true` when the DragItem is hovered over a nested dropzone where the `canDrop` function returned false...

When I use `isOver` instead of `isOverCurrent` on line 33 of Box.js, both drop zones are highlighted when I hover my `DragItem[type=Type1]` over the `Innerbox[type=Type1]`. This is not the desired behaviour, because in that case, only the InnerBox should be highlighted. Hence the use of IsOverCurrent instead of IsOver...

In an ideal world, I could work around the problem by using different drag/drop `ItemTypes` for each `[type=xxx]` value, but as described earlier, the `type` property is completely dynamic, even though I limited it to two Types in this example... So this approach is not possible.

Any thoughts or ideas on how to best approach this?  Is there a way I can somehow set the 'isOver' property to `true` if the child returned canDrop() = false? This already works perfectly for the `drop()` events, but I cant get this to work for the 'hover' state.

TLDR;
How can I make sure a dropzone somehow receives an `isOver() = true` when a child dropzone has returned `canDrop() = false`.

Thanks for reading & thank you very much for your suggestions

**Comment by universse** [comment]
My solution is to use the same type for all drag items and drop targets e.g. 'default'. Then use a different key other than the default 'type' to customise canDrop behavior. Since type is the same for all drag items and drop targets, isOver will be true.
